src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1045,6 +1049,28 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 	foreach(rangeTableCell, rangeTableList)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		bool referenceTable = false;
+
+		if (IsDistributedTable(rangeTableEntry->relid))
```

Comment:
I have removed the distributed table check and also moved the check inside of the `rangeTableEntry->rtekind == RTE_RELATION` check. For that, I had to modify `master_modify_multiple_shards`